tests/test_fastavro.py

```diff
@@ -2105,3 +2106,65 @@ def test_record_named_type():
     records = [{"test1": {"field1": "foo"}, "test2": {"field1": "bar"}}]
     parsed_schema = fastavro.parse_schema(schema)
     assert records == roundtrip(parsed_schema, records)
+
+
+def test_write_required_field_name():
+    '''
+    Test that when we can use tuple style of writing unions
+    (see function `write_union` in `_write`) with primitives
+     not only with records.
+    '''
+
+    schema = {
+        'name': 'test_required',
+        'namespace': 'test',
+        'type': 'record',
+        'fields': [
+                {'name': 'person', 'type': 'string'},
+                {'name': 'age', 'type': 'int'}
+            ]
+    }
+
+    data = [
+        {"person": "James", "age": 34},
+        {"person": "Anthony", "age": None},
+    ]
+
+    new_file = MemoryIO()
+    try:
```

Comment:
Can we use `pytest.raises` here? See `test_no_default` for an example. We should probably also make the match string something like `on field age` rather than just `age`